src/reblox/grid.rs

```diff
@@ -0,0 +1,595 @@
+
+use std::time::Duration;
+
+use pseudo_random::PseudoRandom;
+
+#[derive(PartialEq,Eq,Clone,Copy)]
+pub enum BlockType {
+    I,
+    J,
+    L,
+    O,
+    S,
+    T,
+    Z,
+    GARBAGE,
+    NONE,
+}
+
+const GRID_WIDTH: u8 = 10;
+const GRID_HEIGHT: u8 = 20;
+const START_BLOCK_SPEED_NANOSEC: u32 = 2000000000;
+const END_BLOCK_SPEED_NANOSEC: u32 = 50000000;
+const BLOCK_SPEED_STEP: u32 =
+    (START_BLOCK_SPEED_NANOSEC - END_BLOCK_SPEED_NANOSEC) / 255;
+const LINES_TO_CLEAR_TO_LVL_UP: u8 = 10;
+
+pub fn p2xy(pos: u8) -> (u8, u8) {
+    (pos % GRID_WIDTH, pos / GRID_WIDTH)
+}
+
+pub fn xy2p(x: u8, y: u8) -> u8 {
+    x + y * GRID_WIDTH
+}
+
+pub fn pr2piece(pos: u8, rotation: u8, block_type: BlockType, grid_width: u8) -> [u8;4] {
+    match rotation {
```

Comment:
Perhaps generalize this form instead of one for each rotation (just multiply by `i^rotation`).